sktime/forecasting/compose/_category_compositor.py

```diff
@@ -0,0 +1,479 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements a compositor to utilize forecasters based on ADI/CV categorization."""
+
+__author__ = ["shlok191"]
+
+import copy
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.croston import Croston
+from sktime.forecasting.naive import NaiveForecaster
+from sktime.forecasting.trend import PolynomialTrendForecaster
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.adi_cv import ADICVTransformer
+
+adi_cv_transformer = ADICVTransformer(features=["class"])
+
+
+class CategoryCompositor(BaseForecaster):
```

Comment:
I think the name is a bit non-self-explanatory.

How about `ForecastByGroup`, or `ByGroupForecaster`?

Reply:
I like the sound of `ForecastByGroup`, I'll change the name!